exercises/concept/take-a-number-deluxe/.meta/design.md

```diff
@@ -0,0 +1,41 @@
+# Design
+
+## Learning objectives
+
+- Know what GenServer is.
+- Know how to use GenServer to mix state and behavior.
+- Use the `init` callback to modify initial state.
+- Know the difference between `handle_call`, `handle_cast`, and `handle_info`, and use each.
+- Know how to send user input to callbacks.
+
+## Out of scope
+
+## Concepts
+
+- `genserver`
+
+## Prerequisites
+
+- `processes`
+- `pids`
+- `agent`
+- `behaviours`
```

Comment:
Because GenServer is a behaviour that needs to be implemented, and because its complicated in itself, I think it would be better to teach behaviours separately first. I'll try to think of a story soon.